plasma_framework/test/src/exits/payment/PaymentInFlightExitable.test.js

```diff
@@ -457,5 +457,220 @@ contract('PaymentInFlightExitable', ([_, alice, bob, carol]) => {
+                );
+            });
+
+            it('fails when spending condition is not met', async () => {
+                const newOutputType = OUTPUT_TYPE_ZERO + 1;
+
+                const conditionFalse = await PaymentSpendingConditionFalse.new();
+                await this.exitGame.registerSpendingCondition(
+                    newOutputType, IFE_TX_TYPE, conditionFalse.address,
+                );
+
+                this.challengeArgs.competingTxInputOutputType = newOutputType;
+                await expectRevert(
+                    this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice }),
+                    'Competing input spending condition is not met',
+                );
+            });
+
+            it('fails when competing tx is younger than already known competitor', async () => {
+                // challenge ife as previously
+                await this.framework.setBlock(
+                    this.competingTxBlock.blockNum,
+                    this.competingTxBlock.blockHash,
+                    this.competingTxBlock.blockTimestamp,
+                );
+
+                await this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice });
+
+                // then mine the next block - with the same root hash
+                const nextBlockNum = this.competingTxBlock.blockNum + CHILD_BLOCK_INTERVAL;
+                const nextBlockTimestamp = this.competingTxBlock.blockTimestamp + 1000;
+                const nextCompetitorPos = buildUtxoPos(nextBlockNum, 0, 0);
+
+                await this.framework.setBlock(
+                    nextBlockNum, this.competingTxBlock.blockHash, nextBlockTimestamp,
+                );
+
+                // try to challenge again with competitor from the lastly mined block
+                this.challengeArgs.competingTxPos = nextCompetitorPos;
+                await expectRevert(
+                    this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice }),
+                    'Competing transaction is not older than already known competitor',
+                );
+            });
+
+            it('fails when challenge with the same competing tx twice', async () => {
+                await this.framework.setBlock(
+                    this.competingTxBlock.blockNum,
+                    this.competingTxBlock.blockHash,
+                    this.competingTxBlock.blockTimestamp,
+                );
+
+                await this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice });
+
+                await expectRevert(
+                    this.exitGame.challengeInFlightExitNotCanonical(this.challengeArgs, { from: alice }),
+                    'Competing transaction is not older than already known competitor',
+                );
+            });
+
+            it('succeeds when competitor is not included with appropriate large position', async () => {
```

Comment:
I found it hard to understand this description. Is this testing the case the the competing transaction is inflight (as thus no inclusion proof and no position?

Reply:
yes - I will rephrase in #229